### DISCUSSION TITLE
Unable to Decode Grid Data with the template 5.3

### DISCUSSION THREAD

**Discussion by animus27** [noritada/grib-rs#77] [Q&A] [answered]
Hello, 

I was trying to decode the Grid data from the grib2 file (which can be downloaded from the link below). I am getting `error: ComplexPackingDecodeError(
    NotSupported,
)` when I use gribber. But when I use inspect then it tells me that the templates used are `Templates:
3.0      - Latitude/longitude
4.0      - Analysis or forecast at a horizontal level or in a horizontal layer at a point in time
4.8      - Average, accumulation, extreme values or other statistically processed values at a horizontal level or in a horizontal layer in a continuous or non-continuous time interval
5.3      - Grid point data - complex packing and spatial differencing`

and 5.3 is a supported template. Please help me decode the data from the grib2 file

https://drive.google.com/drive/folders/1ii1D6d37vnk9fRCebkieAxYS31cA8c1r

**Comment by noritada** [comment]
@animus27 Thank you for reporting!

I have already implemented the support but have not merged to the release since it cannot be tested due to the lack of the test data.
I can test with the data you have attached so I can merge the supporting code.

By the way, from the attribute information, the data you provided looks like the data from NCMRWF in India.
Is this data placed on publicly accessible location? If so, I would appreciate it if you could let me know such location.
If not, would you mind if I put (a portion of) the data in the repository of this library?

**Reply by animus27** [reply]
Yes, the data is from NCMRWF in India. Sadly it is not on a publicly accessible location. And I would not mind if you put a potion of or all the data in the repository of this library.

**Reply by noritada** [reply]
Thank you. Okay, I will probably be able to merge and release the supporting version today.

**Reply by animus27** [reply]
BTW the code which you have implemented, is there a branch available on GitHub?

**Reply by noritada** [reply]
It was offline so I've pushed to GitHub.
Now it is available [in this branch](https://github.com/noritada/grib-rs/tree/feat/support-for-first-order-spatial-differencing-in-complex-packing-decoder).

Missing values seem to be correctly handled, but there seems to be a slight divergence in the decoded values compared to when decoded with eccodes. I am investigating.

**Reply by noritada** [reply, marked as answer]
> Missing values seem to be correctly handled, but there seems to be a slight divergence in the decoded values compared to when decoded with eccodes. I am investigating.

It was just that grib-rs and eccodes were reading data from different dates.
Changes has been merged and is now in the master branch.
Please take a look.

**Reply by animus27** [reply]
Thank you very much Kind Sir
